### PR TITLE
Add structured output for `istioctl x precheck`

### PR DESCRIPTION
**Pull request by niwatorrri**
Use Messages type to support structured output for `istioctl x precheck` in a way similar to `istioctl analyze`.

More details in #26220. Code logic has been noticeably changed and should require careful review.

### CODE REVIEW

**Comment by jasonwzm** [comment]
@niwatorrri This is a user visible feature, so we should have release notes added for this PR.

**Comment by esnible** [comment]
@niwatorrri Can you rebase this please?

**Comment by esnible** [comment]
@niwatorrri Do you still want this?  Can you rebase this please, if so?

File: galley/pkg/config/analysis/msg/messages.yaml [hunk 1]
```diff
@@ -329,3 +331,12 @@ messages:
         type: string
       - name: port
         type: string
+
+  - name: "PrecheckFailed"
+    code: IST0201
+    level: Error
+    description: "An error occurred during precheck for istio installation"
```
Comment (esnible) [inline galley/pkg/config/analysis/msg/messages.yaml:338]: "Istio" should be capitalized.

**Review by esnible — COMMENTED** [review]
Do we only want "structured output" or do we want a flag for the old and the new output?

Are people using the old output format?  Do they like it?

**Pull request closed without merging**